presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -212,6 +214,34 @@ private Estimate calculateDistinctValuesCount(Map<String, PartitionStatistics> s
                 DoubleStream::max);
     }
 
+    private Estimate calculateDataSize(Map<String, PartitionStatistics> statisticsByPartitionName, String column)
+    {
+        List<Double> knownPartitionDataSizes = statisticsByPartitionName.values().stream()
+                .map(stats -> {
+                    OptionalDouble averageColumnLength = stats.getColumnStatistics().get(column).getAverageColumnLength();
+                    OptionalLong rowCount = stats.getBasicStatistics().getRowCount();
+                    OptionalLong nullsCount = stats.getColumnStatistics().get(column).getNullsCount();
+                    if (!averageColumnLength.isPresent() || !rowCount.isPresent()) {
+                        return OptionalDouble.empty();
+                    }
+
+                    long nonNullsCount = rowCount.getAsLong() - nullsCount.orElse(0);
+                    return OptionalDouble.of(averageColumnLength.getAsDouble() * nonNullsCount);
+                })
+                .filter(OptionalDouble::isPresent)
+                .map(OptionalDouble::getAsDouble)
+                .collect(toImmutableList());
+
+        double knownPartitionDataSizesSum = knownPartitionDataSizes.stream().mapToDouble(a -> a).sum();
+        long partitionsWithStatsCount = knownPartitionDataSizes.size();
+        long allPartitionsCount = statisticsByPartitionName.size();
+
+        if (partitionsWithStatsCount == 0) {
+            return Estimate.unknownValue();
+        }
+        return new Estimate(knownPartitionDataSizesSum / partitionsWithStatsCount * allPartitionsCount);
```

Comment:
nit: `(knownPartitionDataSizesSum * allPartitionsCount) / partitionsWithStatsCount` in case `knownPartitionDataSizesSum` is very small and `partitionsWithStatsCount` is very big